Add MQT Logo and support Dark-Mode on RtD

SmaranTum opened the pull request:
## Description

This PR adds the official MQT logo to the readme and the documentation. Furthermore, it enables dark-mode support on the RtD page.

## Checklist:

<!---
This checklist serves as a reminder of a couple of things that ensure your pull request will be merged swiftly.
-->

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [x] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

burgholzer commented:
For some reason, and I don't quite know why, the `pyproject.toml` file is messed up on the main branch.

I don't have access to my computer right now, but something must have messed up the file.

SmaranTum commented:
> For some reason, and I don't quite know why, the `pyproject.toml` file is messed up on the main branch.
> 
> I don't have access to my computer right now, but something must have messed up the file.

I added the following to the `pyproject.toml` file:

[project.optional-dependencies]
docs = [
    "sphinx-rtd-dark-mode",
]

I removed it now. lets see if it runs.

burgholzer commented:
Ah. Project metadata for the SyReC project is still specified in the setup.py file and not using pep 621. See, e.g., https://github.com/cda-tum/qmap/pull/171 for how that was introduced in QMAP.

burgholzer commented:
> > For some reason, and I don't quite know why, the `pyproject.toml` file is messed up on the main branch.
> 
> > 
> 
> > I don't have access to my computer right now, but something must have messed up the file.
> 
> 
> 
> I added the following to the `pyproject.toml` file:
> 
> 
> 
> [project.optional-dependencies]
> 
> docs = [
> 
>     "sphinx-rtd-dark-mode",
> 
> ]
> 
> 
> 
> I removed it now. lets see if it runs.
> 
> 
> 
> 

It has to either be added in the setup.py file at the proper place or you have to follow the other suggestion of switching to PEP 621.

SmaranTum commented:
> Ah. Project metadata for the SyReC project is still specified in the setup.py file and not using pep 621. See, e.g., [cda-tum/qmap#171](https://github.com/cda-tum/qmap/pull/171) for how that was introduced in QMAP.

Seems like this is out of this PR's scope. I will edit the PR description and add the logos for this PR now. Will take care of other things in a future PR.

burgholzer commented:
> > Ah. Project metadata for the SyReC project is still specified in the setup.py file and not using pep 621. See, e.g., [cda-tum/qmap#171](https://github.com/cda-tum/qmap/pull/171) for how that was introduced in QMAP.
> 
> 
> 
> Seems like this is out of this PR's scope. I will edit the PR description and add the logos for this PR now. Will take care of other things in a future PR.

Nah. Dark mode can still be enabled. You just have to add the dependency at the proper location in the setup.py file instead of the pyproject.toml file.

The PEP 621 transition should be another PR and would be very welcome.

burgholzer commented:
Also, why is the opengraph dependency from the other PR missing here?! 😩

SmaranTum commented:
> Ah, sorry about that.
